roles/openshift_hosted/tasks/registry/registry.yml

```diff
@@ -1,15 +1,14 @@
 ---
 - name: Retrieve list of openshift nodes matching registry selector
-  command: >
-    {{ openshift.common.client_binary }} --api-version='v1' -o json
-    get nodes -n default --config={{ openshift_hosted_kubeconfig }}
-    --selector={{ openshift.hosted.registry.selector | default('') }}
-  register: registry_nodes_json
-  changed_when: false
+  oc_obj:
+    state: list
+    kind: node
+    selector: "{{ openshift.hosted.registry.selector | default('') }}"
+  register: registry_nodes
   when: openshift.hosted.registry.replicas | default(none) is none
 
 - set_fact:
-    l_node_count: "{{ (registry_nodes_json.stdout | default('{\"items\":[]}') | from_json)['items'] | length }}"
+    l_node_count: "{{ registry_nodes.results.results[0]['items'] | length }}"
```

Comment:
Any reason we couldn't wrap this all in a block and use vars for this instead of set_fact?